tajo-catalog/tajo-catalog-common/src/main/proto/CatalogProtos.proto

```diff
@@ -262,6 +263,18 @@ message PartitionIdentifierProto {
   optional string partitionName = 3;
 }
 
+message PartitionsByAlgebraProto {
+  required string databaseName = 1;
+  required string tableName = 2;
+  required string algebra = 3;
+}
+
+message PartitionsByFilterProto {
+  required string databaseName = 1;
+  required string tableName = 2;
+  required string filter = 3;
```

Comment:
It would be better if you leave a comment of how the filter is serialized into a string.